### ISSUE TITLE
Process stuck with 100% cpu when using jruby 9.4.4.0

### ISSUE THREAD

**Issue by jimtng** [jruby/jruby#7979]
When trying out jruby 9.4.4.0 to run rspec, the process got stuck in a call to `super()` within ruby which refers to a java abstract class parent. This problem doesn't occur when using jruby 9.4.3.0

This is the line that it's stuck on:
https://github.com/openhab/openhab-jruby/blob/2515c24383fa80801de612e759814a1cd8ce4fb0/lib/openhab/core/provider.rb#L123

The parent of that ruby class is [AbstractProvider class](https://github.com/openhab/openhab-core/blob/f2c20af94458371adecd7adc4630312165fc9fed/bundles/org.openhab.core/src/main/java/org/openhab/core/common/registry/AbstractProvider.java#L31) in java

Running `jstack <pid>` gave me this:

```
2023-10-25 20:40:25
Full thread dump OpenJDK 64-Bit Server VM (17.0.9+8-LTS mixed mode, emulated-client, sharing):

Threads class SMR info:
_java_thread_list=0x00007fb74c001750, length=68, elements={
0x00007fb7f0026900, 0x00007fb7f01da0c0, 0x00007fb7f01db4b0, 0x00007fb7f01e0ca0,
0x00007fb7f01e2060, 0x00007fb7f01e3480, 0x00007fb7f01e4a00, 0x00007fb7f01ede60,
0x00007fb7f01fbb20, 0x00007fb7f023e9d0, 0x00007fb7f09b4a70, 0x00007fb7f0aa8130,
0x00007fb7f0c930b0, 0x00007fb7f0cb59b0, 0x00007fb7f0d87b80, 0x00007fb6d01d89f0,
0x00007fb6d01fc7f0, 0x00007fb6d01fdfe0, 0x00007fb6d0222120, 0x00007fb6d02289e0,
0x00007fb67400e270, 0x00007fb67400f260, 0x00007fb6801a4400, 0x00007fb6801af400,
0x00007fb6801b24f0, 0x00007fb6742ad980, 0x00007fb6a0361a20, 0x00007fb6a04217b0,
0x00007fb6a02a2370, 0x00007fb6a0364990, 0x00007fb6a0405df0, 0x00007fb6a03c2410,
0x00007fb6a03c2fc0, 0x00007fb6a03b9180, 0x00007fb6a028f0e0, 0x00007fb648001db0,
0x00007fb68c003760, 0x00007fb68c004870, 0x00007fb648005c60, 0x00007fb634000e70,
0x00007fb6a0727ed0, 0x00007fb6a02c99a0, 0x00007fb6a050f700, 0x00007fb6a02d03d0,
0x00007fb6a0118ac0, 0x00007fb6a0674060, 0x00007fb6a063b6d0, 0x00007fb6a06396b0,
0x00007fb6a0523960, 0x00007fb5f800c6d0, 0x00007fb5f800db70, 0x00007fb5f800ede0,
0x00007fb5f8010060, 0x00007fb6a0739660, 0x00007fb6a07598c0, 0x00007fb6a0a2cf20,
0x00007fb6a0a4e710, 0x00007fb6a0a18740, 0x00007fb6a0a5ec70, 0x00007fb6a03db0e0,
0x00007fb610002890, 0x00007fb6a03de080, 0x00007fb7f0ccda20, 0x00007fb6000014f0,
0x00007fb60c00c350, 0x00007fb64c002910, 0x00007fb61002d560, 0x00007fb74c000d30
}

"main" #1 prio=5 os_prio=0 cpu=28769.94ms elapsed=34.19s tid=0x00007fb7f0026900 nid=0x9632f runnable  [0x00007fb7f7274000]
   java.lang.Thread.State: RUNNABLE
	at org.jruby.java.proxies.ConcreteJavaProxy.isClassOrIncludedPrependedModule(org.jruby.dist/ConcreteJavaProxy.java:399)
	at org.jruby.java.proxies.ConcreteJavaProxy.splitInitialized(org.jruby.dist/ConcreteJavaProxy.java:376)
	at rubyobj.OpenHAB.Core.Items.Metadata.Provider.<init>(/home/openhab/openhab-jruby/lib/openhab/core/items/metadata/provider.rb:226)
	at rubyobj.OpenHAB.Core.Items.Metadata.Provider.<init>(/home/openhab/openhab-jruby/lib/openhab/core/items/metadata/provider.rb:226)
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(java.base@17.0.9/Native Method)
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(java.base@17.0.9/NativeConstructorAccessorImpl.java:77)
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(java.base@17.0.9/DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstanceWithCaller(java.base@17.0.9/Constructor.java:499)
	at java.lang.reflect.Constructor.newInstance(java.base@17.0.9/Constructor.java:480)
	at org.jruby.java.proxies.ConcreteJavaProxy$NewMethodReified.call(org.jruby.dist/ConcreteJavaProxy.java:295)
	at org.jruby.java.proxies.ConcreteJavaProxy$NewMethod.call(org.jruby.dist/ConcreteJavaProxy.java:153)
	at org.jruby.ir.runtime.IRRuntimeHelpers.instanceSuper(org.jruby.dist/IRRuntimeHelpers.java:1319)
	at org.jruby.ir.instructions.InstanceSuperInstr.interpret(org.jruby.dist/InstanceSuperInstr.java:139)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:367)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(org.jruby.dist/InterpreterEngine.java:76)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.INTERPRET_METHOD(org.jruby.dist/InterpretedIRMethod.java:167)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.call(org.jruby.dist/InterpretedIRMethod.java:146)
	at org.jruby.RubyClass.finvokeWithRefinements(org.jruby.dist/RubyClass.java:505)
	at org.jruby.RubyBasicObject.send(org.jruby.dist/RubyBasicObject.java:1661)
	at org.jruby.RubyKernel.send(org.jruby.dist/RubyKernel.java:2348)
	at org.jruby.RubyKernel$INVOKER$s$send.call(org.jruby.dist/RubyKernel$INVOKER$s$send.gen)
	at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodOneOrNBlock.call(org.jruby.dist/JavaMethod.java:423)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:466)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:244)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:318)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(org.jruby.dist/InterpreterEngine.java:76)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.INTERPRET_METHOD(org.jruby.dist/InterpretedIRMethod.java:167)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.call(org.jruby.dist/InterpretedIRMethod.java:146)
	at org.jruby.RubyClass.finvokeWithRefinements(org.jruby.dist/RubyClass.java:505)
	at org.jruby.RubyBasicObject.send(org.jruby.dist/RubyBasicObject.java:1661)
	at org.jruby.RubyKernel.send(org.jruby.dist/RubyKernel.java:2348)
	at org.jruby.RubyKernel$INVOKER$s$send.call(org.jruby.dist/RubyKernel$INVOKER$s$send.gen)
	at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodOneOrNBlock.call(org.jruby.dist/JavaMethod.java:423)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:466)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:244)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:318)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(org.jruby.dist/Interpreter.java:116)
	at org.jruby.runtime.InterpretedIRBlockBody.commonYieldPath(org.jruby.dist/InterpretedIRBlockBody.java:135)
	at org.jruby.runtime.IRBlockBody.doYield(org.jruby.dist/IRBlockBody.java:177)
	at org.jruby.runtime.BlockBody.yield(org.jruby.dist/BlockBody.java:116)
	at org.jruby.runtime.Block.yieldNonArray(org.jruby.dist/Block.java:228)
	at org.jruby.RubyBasicObject.yieldUnder(org.jruby.dist/RubyBasicObject.java:1730)
	at org.jruby.RubyBasicObject.instance_exec(org.jruby.dist/RubyBasicObject.java:2614)
	at org.jruby.RubyBasicObject$INVOKER$i$0$0$instance_exec.call(org.jruby.dist/RubyBasicObject$INVOKER$i$0$0$instance_exec.gen)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:446)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:92)
	at org.jruby.ir.instructions.CallBase.interpret(org.jruby.dist/CallBase.java:548)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:367)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(org.jruby.dist/InterpreterEngine.java:82)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.INTERPRET_METHOD(org.jruby.dist/InterpretedIRMethod.java:204)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.call(org.jruby.dist/InterpretedIRMethod.java:183)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:471)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:259)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:342)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(org.jruby.dist/InterpreterEngine.java:82)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.INTERPRET_METHOD(org.jruby.dist/InterpretedIRMethod.java:204)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.call(org.jruby.dist/InterpretedIRMethod.java:191)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:466)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:244)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:318)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(org.jruby.dist/Interpreter.java:116)
	at org.jruby.runtime.InterpretedIRBlockBody.commonYieldPath(org.jruby.dist/InterpretedIRBlockBody.java:135)
	at org.jruby.runtime.IRBlockBody.doYield(org.jruby.dist/IRBlockBody.java:170)
	at org.jruby.runtime.BlockBody.yield(org.jruby.dist/BlockBody.java:108)
	at org.jruby.runtime.Block.yield(org.jruby.dist/Block.java:189)
	at org.jruby.RubyArray.each(org.jruby.dist/RubyArray.java:1989)
	at org.jruby.RubyArray$INVOKER$i$0$0$each.call(org.jruby.dist/RubyArray$INVOKER$i$0$0$each.gen)
	at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroBlock.call(org.jruby.dist/JavaMethod.java:561)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:446)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:92)
	at org.jruby.runtime.callsite.CachingCallSite.callIter(org.jruby.dist/CachingCallSite.java:103)
	at org.jruby.ir.instructions.CallBase.interpret(org.jruby.dist/CallBase.java:545)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:367)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(org.jruby.dist/InterpreterEngine.java:88)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.INTERPRET_METHOD(org.jruby.dist/InterpretedIRMethod.java:242)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.call(org.jruby.dist/InterpretedIRMethod.java:229)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:476)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:293)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:328)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.INTERPRET_METHOD(org.jruby.dist/InterpretedIRMethod.java:131)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.call(org.jruby.dist/InterpretedIRMethod.java:109)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:446)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:92)
	at org.jruby.runtime.callsite.CachingCallSite.callIter(org.jruby.dist/CachingCallSite.java:103)
	at org.jruby.ir.instructions.CallBase.interpret(org.jruby.dist/CallBase.java:545)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:367)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(org.jruby.dist/Interpreter.java:116)
	at org.jruby.runtime.InterpretedIRBlockBody.commonYieldPath(org.jruby.dist/InterpretedIRBlockBody.java:135)
	at org.jruby.runtime.IRBlockBody.doYield(org.jruby.dist/IRBlockBody.java:170)
	at org.jruby.runtime.BlockBody.yield(org.jruby.dist/BlockBody.java:108)
	at org.jruby.runtime.Block.yield(org.jruby.dist/Block.java:189)
	at org.jruby.ir.runtime.IRRuntimeHelpers.yield(org.jruby.dist/IRRuntimeHelpers.java:495)
	at org.jruby.ir.instructions.YieldInstr.interpret(org.jruby.dist/YieldInstr.java:96)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.processOtherOp(org.jruby.dist/StartupInterpreterEngine.java:155)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:98)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(org.jruby.dist/InterpreterEngine.java:82)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.INTERPRET_METHOD(org.jruby.dist/InterpretedIRMethod.java:204)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.call(org.jruby.dist/InterpretedIRMethod.java:183)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:471)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:259)
	at org.jruby.runtime.callsite.CachingCallSite.callIter(org.jruby.dist/CachingCallSite.java:270)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:341)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(org.jruby.dist/InterpreterEngine.java:82)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.INTERPRET_METHOD(org.jruby.dist/InterpretedIRMethod.java:204)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.call(org.jruby.dist/InterpretedIRMethod.java:191)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:466)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:244)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:318)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(org.jruby.dist/InterpreterEngine.java:88)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.INTERPRET_METHOD(org.jruby.dist/InterpretedIRMethod.java:242)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.call(org.jruby.dist/InterpretedIRMethod.java:229)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:476)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:293)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:328)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.INTERPRET_METHOD(org.jruby.dist/InterpretedIRMethod.java:131)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.call(org.jruby.dist/InterpretedIRMethod.java:109)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:446)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:92)
	at org.jruby.ir.instructions.CallBase.interpret(org.jruby.dist/CallBase.java:548)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:367)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(org.jruby.dist/InterpreterEngine.java:76)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.INTERPRET_METHOD(org.jruby.dist/InterpretedIRMethod.java:167)
	at org.jruby.internal.runtime.methods.InterpretedIRMethod.call(org.jruby.dist/InterpretedIRMethod.java:154)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:456)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:195)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:350)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.Interpreter.INTERPRET_ROOT(org.jruby.dist/Interpreter.java:96)
	at org.jruby.ir.interpreter.Interpreter.execute(org.jruby.dist/Interpreter.java:81)
	at org.jruby.ir.interpreter.Interpreter.execute(org.jruby.dist/Interpreter.java:30)
	at org.jruby.ir.IRTranslator.execute(org.jruby.dist/IRTranslator.java:42)
	at org.jruby.Ruby.runInterpreter(org.jruby.dist/Ruby.java:1293)
	at org.jruby.Ruby.loadFile(org.jruby.dist/Ruby.java:2990)
	at org.jruby.runtime.load.LibrarySearcher$ResourceLibrary.load(org.jruby.dist/LibrarySearcher.java:925)
	at org.jruby.runtime.load.LibrarySearcher$FoundLibrary.load(org.jruby.dist/LibrarySearcher.java:883)
	at org.jruby.runtime.load.LoadService.load(org.jruby.dist/LoadService.java:383)
	at org.jruby.RubyKernel.loadCommon(org.jruby.dist/RubyKernel.java:1137)
	at org.jruby.RubyKernel.load(org.jruby.dist/RubyKernel.java:1107)
	at org.jruby.RubyKernel$INVOKER$s$load.call(org.jruby.dist/RubyKernel$INVOKER$s$load.gen)
	at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodOneOrNBlock.call(org.jruby.dist/JavaMethod.java:423)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(org.jruby.dist/CachingCallSite.java:466)
	at org.jruby.runtime.callsite.CachingCallSite.call(org.jruby.dist/CachingCallSite.java:244)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(org.jruby.dist/InterpreterEngine.java:318)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(org.jruby.dist/StartupInterpreterEngine.java:66)
	at org.jruby.ir.interpreter.Interpreter.INTERPRET_ROOT(org.jruby.dist/Interpreter.java:96)
	at org.jruby.ir.interpreter.Interpreter.execute(org.jruby.dist/Interpreter.java:81)
	at org.jruby.ir.interpreter.Interpreter.execute(org.jruby.dist/Interpreter.java:30)
	at org.jruby.ir.IRTranslator.execute(org.jruby.dist/IRTranslator.java:42)
	at org.jruby.Ruby.runInterpreter(org.jruby.dist/Ruby.java:1315)
	at org.jruby.Ruby.runInterpreter(org.jruby.dist/Ruby.java:1319)
	at org.jruby.Ruby.runNormally(org.jruby.dist/Ruby.java:1202)
	at org.jruby.Ruby.runNormally(org.jruby.dist/Ruby.java:1179)
	at org.jruby.Ruby.runNormally(org.jruby.dist/Ruby.java:1215)
	at org.jruby.Ruby.runFromMain(org.jruby.dist/Ruby.java:993)
	at org.jruby.Main.doRunFromMain(org.jruby.dist/Main.java:398)
	at org.jruby.Main.internalRun(org.jruby.dist/Main.java:282)
	at org.jruby.Main.run(org.jruby.dist/Main.java:227)
	at org.jruby.Main.main(org.jruby.dist/Main.java:199)

"Reference Handler" #2 daemon prio=10 os_prio=0 cpu=18.71ms elapsed=34.18s tid=0x00007fb7f01da0c0 nid=0x96337 waiting on condition  [0x00007fb7b31a0000]
   java.lang.Thread.State: RUNNABLE
	at java.lang.ref.Reference.waitForReferencePendingList(java.base@17.0.9/Native Method)
	at java.lang.ref.Reference.processPendingReferences(java.base@17.0.9/Reference.java:253)
	at java.lang.ref.Reference$ReferenceHandler.run(java.base@17.0.9/Reference.java:215)

"Finalizer" #3 daemon prio=8 os_prio=0 cpu=61.22ms elapsed=34.18s tid=0x00007fb7f01db4b0 nid=0x96338 in Object.wait()  [0x00007fb7b2f9f000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(java.base@17.0.9/Native Method)
	- waiting on <no object reference available>
	at java.lang.ref.ReferenceQueue.remove(java.base@17.0.9/ReferenceQueue.java:155)
	- locked <0x0000000415801488> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(java.base@17.0.9/ReferenceQueue.java:176)
	at java.lang.ref.Finalizer$FinalizerThread.run(java.base@17.0.9/Finalizer.java:172)

"Signal Dispatcher" #4 daemon prio=9 os_prio=0 cpu=0.23ms elapsed=34.18s tid=0x00007fb7f01e0ca0 nid=0x96339 waiting on condition  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Service Thread" #5 daemon prio=9 os_prio=0 cpu=16.26ms elapsed=34.18s tid=0x00007fb7f01e2060 nid=0x9633a runnable  [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE


<snipped>

"VM Periodic Task Thread" os_prio=0 cpu=9.69ms elapsed=34.12s tid=0x00007fb7f0028620 nid=0x96343 waiting on condition  

JNI global refs: 15, weak refs: 0

```

The full output is uploaded as an attachment.
[jruby-stack.txt](https://github.com/jruby/jruby/files/13165505/jruby-stack.txt)

**Comment by headius** [comment]
Thank you for the report! I don't know offhand what changed to cause this, but one possible candidate is this improvement by @kares: https://github.com/jruby/jruby/commit/bd9d40c02a1a693b46aa14e5feac7d64b5442246

That patch modifies or introduces the top two methods in your trace. At a glance I can't tell whether it broke something. @kares Any thoughts here?

@jimtng You might try bisecting around that commit and see if you can confirm it.

**Comment by jimtng** [comment]
To reproduce, make sure java17 is set as default, then
```
git clone https://github.com/openhab/openhab-jruby
cd openhab-jruby
bundle install
bundle exec rake openhab:setup
bin/rspec # this will trigger the issue
```

Note `bundle exec rake openhab:setup` will download openhab into `openhab-jruby/tmp/openhab`, start it, configure it, then stop it.
After testing, the whole `openhab-jruby` dir can be deleted, it shouldn't leave any other trace on the system.

**Comment by headius** [comment]
@jimtng I must have missed a step:

```
Listening for transport dt_socket at address: 5005
printenv: illegal option -- 0
usage: printenv [name]
printenv: illegal option -- 0
usage: printenv [name]
#<KeyError: key not found: "OPENHAB_USERDATA"
Did you mean?  "OPENHAB_HOME">
org/jruby/RubyHash.java:1429:in `fetch'
/Users/headius/work/jruby/openhab-jruby/lib/openhab/rspec/karaf.rb:706:in `felix_cm'
/Users/headius/work/jruby/openhab-jruby/lib/openhab/rspec/karaf.rb:716:in `cleanup_clone'
/Users/headius/work/jruby/openhab-jruby/lib/openhab/rspec/karaf.rb:710:in `cleanup_instance'
/Users/headius/work/jruby/openhab-jruby/lib/openhab/rspec/karaf.rb:118:in `start_instance'
/Users/headius/work/jruby/openhab-jruby/lib/openhab/rspec/karaf.rb:58:in `launch'
/Users/headius/work/jruby/openhab-jruby/lib/openhab/rspec/helpers.rb:216:in `launch_karaf'
/Users/headius/work/jruby/openhab-jruby/lib/openhab/rspec/hooks.rb:22:in `<module:RSpec>'
/Users/headius/work/jruby/openhab-jruby/lib/openhab/rspec/hooks.rb:11:in `<module:OpenHAB>'
/Users/headius/work/jruby/openhab-jruby/lib/openhab/rspec/hooks.rb:3:in `<main>'
```

**Comment by jimtng** [comment]
Did you run: `bundle exec rake openhab:setup` using a working ruby (i.e. not jruby 9.4.4.0. jruby-9.4.3.0 or mri 3.1.x should work)

**Comment by headius** [comment]
Ah, no I ran it all with JRuby master. I'll run that step with 9.4.3.0 instead.

**Comment by headius** [comment]
I'm still getting the same error. Some env var you must have on your system?

```
An error occurred while loading spec_helper.
Failure/Error: @felix_cm ||= use_root_instance ? ENV.fetch("OPENHAB_USERDATA") : "#{path}/config"

KeyError:
  key not found: "OPENHAB_USERDATA"
  Did you mean?  "OPENHAB_HOME"
# ./lib/openhab/rspec/karaf.rb:706:in `felix_cm'
```

**Comment by jimtng** [comment]
The environment vars should be loaded from `openhab-jruby/tmp/openhab/runtime/setenv`, so the shell doesn't need to have it.

Hmm, yes, I am getting the same error when trying it on a MacOS. 

It failed because shell.rb calls `printenv -0` and this only works on Linux. MacOS's printenv doesn't recognise this argument.

Could you try it out on Linux?

**Comment by ccutrer** [comment]
> Could you try it out on Linux?

... or just update your checkout of openhab-jruby; I just fixed that issue (though I still haven't run the full suite of rspec on Mac, so there might be more...)

**Comment by jimtng** [comment]
I did try it on Mac with the new fix and it works now.

**Comment by kares** [comment]
Managed to reproduce in isolation, I see that the loop doesn't have a way to end in this case (seems I missed a line where refactoring).

**Comment by headius** [comment]
I think this warrants a 9.4.5.0 sooner than later, or if we are risk averse we pull this and one or two other simple critical fixes into a 9.4.4.1.
